source/transactions-convenient-api/tests/commit-transienttransactionerror.yml

```diff
@@ -0,0 +1,202 @@
+database_name: &database_name "withTransaction-tests"
+collection_name: &collection_name "test"
+
+data: []
+
+# These tests use error codes where the TransientTransactionError label will be
+# applied to the error response for commitTransaction. This will cause the
+# entire transaction to be retried instead of commitTransaction.
+#
+# See: https://github.com/mongodb/mongo/blob/r4.1.6/src/mongo/db/handle_request_response.cpp
+tests:
+  -
+    description: transaction is retried commitTransaction TransientTransactionError (NotMaster)
+    failPoint:
+      configureFailPoint: failCommand
+      mode: { times: 1 }
+      data:
+          failCommands: ["commitTransaction"]
+          errorCode: 10107 # NotMaster
+          closeConnection: false
+    operations:
+      -
+        name: withTransaction
+        object: session0
+        callback: &callback
+          operations:
+            -
+              name: insertOne
+              object: collection
+              arguments:
+                session: session0
+                document: { _id: 1 }
+              result:
+                insertedId: 1
+        result:
```

Comment:
There shouldn't be a result here, as the transaction ultimately succeeds and therefore any transient errors are swallowed.  Other tests in this file have the same problem.

Reply:
Acknowledged in https://github.com/mongodb/specifications/pull/435#discussion_r245136611. Will fix.